content/docs/user-guide/gems/reference/aws/aws-gamelift/advanced-topics.md

```diff
@@ -0,0 +1,118 @@
+---
+linkTitle: Advanced Topics
+title: AWS GameLift Gem Advanced Topics
+description: Learn some advanced ways to use the AWS GameLift Gem in Open 3D Engine (O3DE).
+toc: true
+---
+
+This topic covers some advanced ways to use the **AWS GameLift** Gem. 
+
+
+## Use resource mapping files to look up fleets
+
+A *fleet* is a set of Amazon EC2 virtual machines, called *instances*, that represents your hosting resources. A *fleet ID* is a unique identifier for each fleet, which Amazon GameLift generates. 
+
+In most cases, your game will use more than one fleet. To avoid using a fixed fleet ID in your script, you can use a resource mapping file, which lets you look up fleet IDs at runtime. Using a resource mapping file helps keep your code or script clean and consistent. For more information, refer to [Resource Mapping Files](/docs/user-guide/gems/reference/aws/aws-core/resource-mapping-files/).
+
+The following examples use [Script Canvas](/docs/user-guide/scripting/script-canvas/), but you can also search and join sessions using the [AWS GameLift Gem C++ API](cpp-api/). 
+
+
+### Example: Fleet lookup
+
+This example script shows you how to look up a fleet from the resource mapping file. Then, with the fleet, you create a session that allows one maximum player. (You can also do this using the AWS GameLift Gem C++ API.)
+
+In Script Canvas, follow these steps:
+
+1. Get the request object for creating a session by using the **Get request** node.
+
+2. Look up the fleet with the ID `MyGameLiftFleetId` by using the **GetResourceNameId** node. 
+
+3. For the corresponding fleet, set the maximum number of players to be `1` by using the **FleetId** and **MaxPlayer** nodes. 
+
+4. Create a session on the fleet by using the **CreateSessionAsync** node. 
+
+![A script that gets a fleet from a resource mapping file and creates a session with 1 maximum player.](/images/user-guide/gems/reference/aws/aws-gamelift/createsessionandresourcemapping.PNG)
+
+
+### Example: Multiple fleet lookup
+
+This example script shows you how to look up multiple fleets from the resource mapping file. If you have multiple fleets, you can store the fleet IDs in an array. Then, you can iterate through each fleet and search for active sessions in the corresponding fleet. (You can also do this using the AWS GameLift Gem C++ API.)
```

Comment:
This example script shows you how to search active game session across multiple fleets by using resource mapping file.